GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Gas.cs

```diff
@@ -0,0 +1,98 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+// control gas change of the car
+public class Gas : MonoBehaviour
+{
+    public GameObject noGasText;
+    public GameObject gasLevelImageObj;
+
+    private const float FullGasLevel = 5.0f;
+    private const float Mpg = 0.1f;
+    private float _gasLevel = 5.0f;
+    private float _droveDistanceBeforeLastFill = 0f;
+    private float _totalDistanceDriven = 0f;
+    private Image _gasLevelImage;
+    private readonly Color32 _darkRedColor = new Color32(196, 92, 29, 255);
+    private readonly Color32 _orangeColor = new Color32(255, 196, 0, 255);
+    private readonly Color32 _lightGreenColor = new Color32(125, 210, 76, 255);
+    private readonly Color32 _greenColor = new Color32(94, 201, 93, 255);
+    private const float LowVolumeCoefficient = 0.2f;
+    private const float MediumVolumeCoefficient = 0.4f;
+    private const float HighVolumeCoefficient = 0.6f;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        _gasLevelImage = gasLevelImageObj.GetComponent<Image>();
+    }
+
+    // check if there is gas left in the tank
+    public bool HasGas()
+    {
+        if (_gasLevel > 0)
+        {
+            return true;
+        }
+        else
+        {
+            noGasText.SetActive(true);
+            return false;
+        }
+    }
+
+    // reset the gas level and distance when fill the gas
```

Comment:
the distance here is the distance the player has driven before the last time fill gas. I need to reset it since the way I compute gas consumption is (total distance - distance before last time fill).